src/global/variables/clauses/IntegerLiteral.java

```diff
@@ -1,26 +1,81 @@
 package global.variables.clauses;
 
+import global.tools.InvalidClauseException;
+import global.utils.RegexUtil;
 import global.variables.Clause;
 
 public class IntegerLiteral extends Clause {
-    int value;
+    int value = 0;
+    int upper;
+    int lower;
 
     public IntegerLiteral(int value) {
         super();
         this.value = value;
+        this.upper = value;
+        this.lower = value;
+
     }
 
     public IntegerLiteral(int value, String name) {
         super(name);
         this.value = value;
+        this.upper = value;
+        this.lower = value;
+    }
+
+    public IntegerLiteral(int lower, int upper) {
+        super();
+        this.lower = lower;
+        this.upper = upper;
+    }
+
+    public IntegerLiteral(int lower, int upper, String name) {
+        super(name);
+        this.lower = lower;
+        this.upper = upper;
+    }
+
+    public int getUpper() {
+        return upper;
+    }
+
+    public int getLower() {
+        return lower;
     }
 
     public int getValue() {
         return value;
     }
 
+    @Override
+    public boolean validate(String matchGroupString) {
+        if (getLower() != getUpper()) {
+            int matchGroupInt = Integer.parseInt(matchGroupString);
+            return matchGroupInt >= getLower() && matchGroupInt <= getUpper();
+        }
+        return true;
+    }
+
+    @Override
+    public void validateParams() throws InvalidClauseException {
+        if (getLower() != getUpper()) {
+            if (getLower() > getUpper()) {
+                throw new InvalidClauseException("Lower bound cannot be greater than upper bound");
+            }
+        }
+    }
+
+    public String getValueOrRegexRange() {
+        if (getLower() == getUpper()) {
+            return String.valueOf(getValue());
```

Comment:
I think the logic here should just be "if upper and lower are defined, use them for regex, if they're not and there is a defined value, use that instead"